src/index.ts

```diff
@@ -272,15 +272,18 @@ export async function main(args: CliArguments): Promise<void> {
       process.exit(1);
     }
   } catch (error) {
+    const errorWithPath = error as Error & { path?: string };
```

Comment:
That's true. Happy to make that change if it's preferred.

Reply:
Simply casting the error doesn't fix anything. Either we should enable that flag, or we should verify that the property exists before accessing it. So asserting the type, instead of casting it.

I prefer asserting over the flag. Can you update the pull to do so?

Reply:
You can't check if the property exists on type `unknown` (or even on type `Error`). This only worked previously because TypeScript was casting the error as the very permissive `any` type. So as I see it there are three options:

1. Use `instanceof` to deal with specific errors (including the one that has the `path` property). I like this option because it aligns with the intention of the TypeScript rule. Since you can throw with literally anything, we can't assume anything about the type. It is `unknown`. Although I like this one the best, it does require the most refactoring.

2. Disable the flag @AKK9 mentioned and continue to treat the caught error as `any` instead of as `unknown`.

3. Cast the error as what we expect it to be. I'd say `Error & { path?: string }` is better than `any` which is what TypeScript was doing behind the scenes before.

I didn't initially go with option 1 because I didn't want to expand the scope of this pull request too much. Let me know how you'd like to proceed though. I'm happy to implement any of these options.

Reply:
Yeah, 1 is pretty much what I meant. You worded it better. Casting something to a specific type, is not much better then using `any` in my experience. If we do this, better make it right. 1 has my preference 🙂